components/Common/ShellBox/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import styles from './index.module.scss';
+import ReactDomServer from 'react-dom/server';
+
+interface Props {
+  children: React.ReactNode | undefined;
+  textToCopy?: string;
+}
+
+const ShellBox = ({ children, textToCopy }: React.PropsWithChildren<Props>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const handleCopyCode = useCallback(
+    async (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+      const text = textToCopy
+        ? textToCopy
+        : ReactDomServer.renderToString(<>children</>);
```

Comment:
Usage of `react-dom/server` should be avoided in client side since it will increase the bundle size of the app

[React DOCS](https://react.dev/reference/react-dom/server/renderToString#removing-rendertostring-from-the-client-code)